tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -762,18 +762,11 @@ def touch_home_button(self):
             homescreen.wait_to_be_displayed()
             apps.switch_to_displayed_app()
         else:
-            apps.switch_to_displayed_app()
-            mode = self.marionette.find_element(By.TAG_NAME, 'body').get_attribute('class')
             self._dispatch_home_button_event()
             apps.switch_to_displayed_app()
-            if 'edit-mode' in mode:
-                # touching home button will exit edit mode
-                Wait(self.marionette).until(lambda m: m.find_element(
-                    By.TAG_NAME, 'body').get_attribute('class') != mode)
-            else:
-                # touching home button inside homescreen will scroll it to the top
-                Wait(self.marionette).until(lambda m: m.execute_script(
-                    "return window.wrappedJSObject.scrollY") == 0)
+
+            # touching home button inside homescreen will scroll it to the top
+            Wait(self.marionette).until(homescreen.scoll_position['y'] == 0)
```

Comment:
We can improve these 2 lines by defining another method in Homescreen:
```py
Wait(self.marionette).until(homescreen.is_at_the_topmost_position)
```

Having no hardcoded value in this function will help to maintain, if one day the topmost position is not 0 anymore